gap/matrix/classical.gi

```diff
@@ -1701,22 +1707,36 @@ RECOG.NonGenericOrthogonalPlus := function(recognise,grp)
         fi;
         pgrp := ProjectiveActionOnFullSpace( grp, recognise.field, d );
         orbs := Orbits( pgrp, MovedPointsPerms( GeneratorsOfGroup(pgrp)));
-        if Set(List(orbs, Length)) <> [ 1080, 1120] then
+        if Set(orbs, Length) <> [ 1080, 1120 ] then
              recognise.isSOContained := false;
              return false;
         fi;
-        if Size(pgrp) mod 4952179814400 = 0 then
+        if Size(pgrp) mod 4952179814400 = 0 then # compare to Size(POmega(+1,8,3))
              return CheckFlag();
         else
              recognise.isSOContained := false;
              return false;
         fi;
     elif d = 8 and  q =  5 then
-        #ACN Feb 07: added fix - need also elements of order a mult of 3
-        #            these are missing in the paper [NP99]
-        if not HasElementsMultipleOf( recognise.orders, [7,13,3])  then
+        ## 2.July.2019: There is a mistake in the paper here
+        ## There are maximal subgroups of Omega+(8,5) that contain
+        ## elements of order 7,13,3
+        if not HasElementsMultipleOf( recognise.orders, [7,13,3,312])  then
             return fail;
         fi;
+
+        pgrp := ProjectiveActionOnFullSpace( grp, recognise.field, d );
```

Comment:
On my laptop I get:
```
gap> ProjectiveActionOnFullSpace( Omega(+1,8,5), GF(5), 8);; time;
648
```
So this takes more than half a second. That seems rather slow. We also may end up re-computing this action later on... As it is, it caused a severe slowdown of our test suite, because now testing with the input `O(+1,8,5)` is very slow. Before this PR:
```
gap> ReadPackage("recog", "tst/naming.g");
gap> TestNaming("SO", +1, 8, 5); time;
645
```
With this PR:
```
gap> ReadPackage("recog", "tst/naming.g");
gap> TestNaming("SO", +1, 8, 5); time;
212476
```

So it goes in total from 0.6 seconds to 3.5 minutes!!!

Do we really need the whole orbit?  Perhaps we can instead just check the length of the orbit of a single vector or so?

Reply:
I just noticed that we do the same computation in other cases too. So is this slower because the dimension is bigger or do we maybe not test the other cases?

Reply:
The other cases are indeed significantly smaller; just computing the orbit takes some time. Moreover 5^7 > 2^16 so "big" permutations have to be used; 4*5^7 = 320kb per permutation kills most caches quickly.

Reply:
@aniemeyer and I are looking at this.

I'll use magma to compute the maximal subgroups of GO+(8,5). Then we can see how to distinguish them from the Omega+(8,5).

Reply:
Also, we should write down which theorem this code is based upon.

Reply:
Some thoughts:
- It should be possible to compute only one orbit. I've tried it and unfortunately that also takes half a second. 
- I haven't figured out yet how to compute the maximal sugroups of GO+(8,5) in Magma. I can get the maximal ones of Omega+(8,5), but that's it.

Reply:
Do you know by now what the relevant "bad" maximal subgroup is?

Also, I noticed that while I complained that `pgrp := ProjectiveActionOnFullSpace(...)` takes half a second, the call `Size(pgrp)` later actually takes *5 seconds* ... So yeah, we need to work on this :-)

Reply:
```
gap> orb:=Orbit(G,One(G)[1], OnLines);; time;
90
gap> Length(orb);
19656
gap> pg:=Action(G,orb,OnLines);; time;
223
gap> Size(pg2); time;
8911539000000000000
1369
```

So perhaps this would work:
```gap
orb:=Orbit(grp, One(grp)[1], OnLines);
if not Length(orb) in [ 19656, 39000 ] then  ... return false; fi
pgrp:=Action(G,orb,OnLines);
if Size(pgrp) ...
```
Of course `pgrp` may underestimate the actual size of the group. In order to understand whether the above is sufficient or not, I'd need to better understand what the group is we want to distinguish. @aniemeyer just told me that it is O(7,5) which is of course muuuch smaller; it has size 914004000000000. So perhaps it suffices to test whether `grp` resp. `pgrp` has more than 914004000000000 elements; for this, one could abort the random schreier sims early as soon as one has a group bigger than that.

Reply:
@fingolfin Oh, interesting! Aborting the schreier sims computation then seems like the way to go. Cheryl and Alice had a chat about how to do this case last week (I think) and we'll try to put it into code after the GAP days.